pest/src/iterators/pairs.rs

```diff
@@ -167,6 +167,15 @@ impl<'i, R: RuleType> Pairs<'i, R> {
         tokens::new(self.queue, self.input, self.start, self.end)
     }
 
+    #[inline]
+    pub fn peek(&mut self) -> Option<<Self as Iterator>::Item> {
```

Comment:
Why use `<Self as Iterator>::Item` instead of `Pair`?

```suggestion
    pub fn peek(&mut self) -> Option<Pair> {
```